src/lib/tuning/greedy_selector.cpp

```diff
@@ -0,0 +1,143 @@
+#include "greedy_selector.hpp"
+
+#include <algorithm>
+#include <limits>
+#include <list>
+
+#include "utils/logging.hpp"
+
+namespace opossum {
+
+// This subroutine scans the least desirable available choices until the
```

Comment:
The comment says the least desirable available choices are scanned and you start with `sorted_choices.begin()`. Shouldn't `begin` point to the most beneficial choice according to the comment above: `so that the most beneficial operations come before less useful operations`

Reply:
You're mixing two different things: The comment you mentioned refers to the **operation sequence** resulting from the selection step, whereas this comment (and function) deal with the sorted list of available **choices**.
Line 54 contains a description of how this is sorted.